llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -2762,7 +2766,46 @@ bool AMDGPULegalizerInfo::buildPCRelGlobalAddress(Register DstReg, LLT PtrTy,
   if (PtrTy.getSizeInBits() == 32)
     B.buildExtract(DstReg, PCReg, 0);
   return true;
- }
+}
+
+// Emit a ABS32_LO / ABS32_HI relocation stub.
```

Comment:
Why do this here instead of selection of the address? The G_GLOBAL_VALUE is more useful for post-legalize combines

Reply:
You mean, as part of the IRTranslator which builds the G_GLOBAL_VALUE? Sure, that would work somehow, but why would we have relocation logic at two different places? I intended to have the GISel implementation next to the pc-rel relocation support, that's why I implemented it here.

Reply:
I think Matt meant to have this as part of instruction selection which runs after legalization and register bank allocation. It's a fair question, but I agree with you that it seems plausible to have all G_GLOBAL_VALUE handling in a single location.